cli/module_loader.rs

```diff
@@ -692,40 +713,74 @@ impl NpmModuleLoader {
     }
   }
 
-  pub fn resolve_nv_ref(
+  pub fn resolve_req_reference(
     &self,
-    nv_ref: &NpmPackageNvReference,
+    req_ref: &NpmPackageReqReference,
     permissions: &PermissionsContainer,
+    referrer: &ModuleSpecifier,
   ) -> Result<ModuleSpecifier, AnyError> {
     let package_folder = self
       .npm_resolver
-      .resolve_pkg_folder_from_deno_module(nv_ref.nv())?;
+      .resolve_pkg_folder_from_deno_module_req(req_ref.req(), referrer)?;
     self
-      .handle_node_resolve_result(self.node_resolver.resolve_npm_reference(
+      .resolve_package_sub_path(
         &package_folder,
-        nv_ref.sub_path(),
-        NodeResolutionMode::Execution,
+        req_ref.sub_path(),
         permissions,
-      ))
-      .with_context(|| format!("Could not resolve '{}'.", nv_ref))
+      )
+      .with_context(|| format!("Could not resolve '{}'.", req_ref))
   }
 
-  pub fn resolve_req_reference(
+  fn resolve_package_sub_path(
     &self,
-    req_ref: &NpmPackageReqReference,
+    package_folder: &Path,
+    sub_path: Option<&str>,
     permissions: &PermissionsContainer,
   ) -> Result<ModuleSpecifier, AnyError> {
-    let package_folder = self
-      .npm_resolver
-      .resolve_pkg_folder_from_deno_module_req(req_ref.req())?;
-    self
-      .handle_node_resolve_result(self.node_resolver.resolve_npm_reference(
-        &package_folder,
-        req_ref.sub_path(),
-        NodeResolutionMode::Execution,
-        permissions,
-      ))
-      .with_context(|| format!("Could not resolve '{}'.", req_ref))
+    self.handle_node_resolve_result(self.node_resolver.resolve_npm_reference(
+      package_folder,
+      sub_path,
+      NodeResolutionMode::Execution,
+      permissions,
+    ))
+  }
+
+  fn handle_node_resolve_result(
+    &self,
+    result: Result<Option<NodeResolution>, AnyError>,
+  ) -> Result<ModuleSpecifier, AnyError> {
+    let response = match result? {
+      Some(response) => response,
+      None => return Err(generic_error("not found")),
```

Comment:
Could we add some error context here, I've hit "not found" error a few times and it's really cryptic what's going on. Feel free to address it in a follow up

Reply:
Will do so in a follow-up after byonm sometime. This method was just moved here from below. I don't want to make any real modifications at this point.